generic/MAPL_Generic.F90

```diff
@@ -4955,16 +4960,25 @@ recursive integer function MAPL_AddChildFromDSO(NAME, userRoutine, grid, ParentG
   call ESMF_VMGet(vm, mpiCommunicator=comm, __RC__)
   CHILD_META%t_profiler = TimeProfiler(trim(NAME), comm_world=comm)
 
+  ! Construct a valid shared object library name
+
+  ! 1. Get the basename of the file without the extension
+  shared_object_library_basename = get_file_basename(SharedObj)
+  ! 2. Get the system dso suffix
+  system_dso_suffix = get_system_dso_suffix()
+  ! 3. Make a new shared object name
+  shared_object_library_with_system_dso_suffix = trim(shared_object_library_basename)//trim(system_dso_suffix)
+
   t_p => get_global_time_profiler()
   call t_p%start(trim(NAME),__RC__)
   call CHILD_META%t_profiler%start(__RC__)
   call CHILD_META%t_profiler%start('SetService',__RC__)
   gridcomp => META%GET_CHILD_GRIDCOMP(I)
-  call ESMF_GridCompSetServices ( gridcomp, userRoutine, sharedObj=sharedObj,__RC__)
+  call ESMF_GridCompSetServices ( gridcomp, userRoutine, sharedObj=shared_object_library_with_system_dso_suffix,__RC__)
```

Comment:
Don't we need a `userRC` here?

Reply:
Oh. I suppose so. I'll have to learn how to do that.

Reply:
It should be fine with this
call ESMF_GridCompSetServices ( gridcomp, userRoutine, sharedObj=shared_object_library_with_system_dso_suffix, userRC=userRC, __RC__)
_VERIFY(userRC)

Reply:
oops, __RC__) should still be there without being formatted here

Reply:
_ _ RC _ _

Reply:
Fixed.